website/content/docs/sync/gcpsm.mdx

```diff
@@ -0,0 +1,184 @@
+---
+layout: docs
+page_title: Google Cloud Platform Secret Manager - Secrets Sync Destination
+description: The Google Cloud Platform Secret Manager destination syncs secrets from Vault to GCP.
+---
+
+# Google Cloud Platform Secret Manager sync destination
+
+The Google Cloud Platform Secret Manager sync destination allows Vault to safely synchronize secrets to your GCP projects.
+This is a low footprint option that enables your applications to benefit from Vault-managed secrets without requiring them
+to connect directly with Vault. This guide walks you through the configuration process.
+
+Prerequisites:
+* Ability to read or create KVv2 secrets
+* Ability to create GCP Service Account credentials with access to the Secret Manager
+* Ability to create sync destinations and associations on your Vault server
+
+## Setup
+
+1. If you do not already have a Service Account, navigate to the IAM & Admin page in the Google Cloud console to
+[create a new Service Account](https://cloud.google.com/iam/docs/service-accounts-create) with the
+[necessary permissions](/vault/docs/sync/gcpsm#permissions). [Instructions](/vault/docs/sync/gcpsm#provision-service-account)
+to provision this Service Account via Terraform can be found below.
+
+1. Configure a sync destination with the Service Account json credentials created in the previous step. See docs for
+[alternative ways](/vault/docs/secrets/gcp#authentication) to pass in the `credentials` parameter.
+
+  ```shell-session
+  $ vault write sys/sync/destinations/gcp-sm/my-dest credentials=@path/to/credentials.json
+  Key                   Value
+  ---                   -----
+  connection_details    map[credentials:*****]
+  name                  my-dest
+  type                  gcp-sm
+  ```
+
+## Usage
+
+1. If you do not already have a KVv2 secret to sync, mount a new KVv2 secrets engine and create a secret.
+
+  ```shell-session
+  $ vault secrets enable -path=my-kv kv-v2
+  Success! Enabled the kv-v2 secrets engine at: my-kv/
+
+  $ vault kv put -mount=my-kv my-secret foo=bar
+  ==== Secret Path ====
+  my-kv/data/my-secret
+
+  ======= Metadata =======
+  Key                Value
+  ---                -----
+  created_time       <timestamp>
+  custom_metadata    <nil>
+  deletion_time      n/a
+  destroyed          false
+  version            1
+  ```
+
+1. Create an association between your GCP SM destination and the secret to synchronize
+
+  ```shell-session
+  vault write sys/sync/destinations/aws-sm/my-dest/associations/set mount=my-kv secret_name=my-secret
```

Comment:
Sorry I know this is not in review, but found this while going through the flow and setting things up
```suggestion
  vault write sys/sync/destinations/gcp-sm/my-dest/associations/set mount=my-kv secret_name=my-secret
```

Reply:
Oh nice find, definitely missed that! 

Feel free to comment even if it is in Draft. We didn't put it for review only because some parts are not completed yet.